src/storage/src/hummock/event_handler/hummock_event_handler.rs

```diff
@@ -368,7 +377,130 @@ impl HummockEventHandler {
         });
     }
 
-    fn handle_version_update(&mut self, version_payload: Payload) {
+    async fn fill_cache(&mut self, delta: &HummockVersionDelta) -> HummockResult<()> {
+        let stats = StoreLocalStatistic::default();
+        for (_group_id, group_delta) in &delta.group_deltas {
+            for d in &group_delta.group_deltas {
+                let mut flatten_reqs = vec![];
+                if let Some(group_delta::DeltaType::IntraLevel(level_delta)) = d.delta_type.as_ref()
+                {
+                    for sst in &level_delta.inserted_table_infos {
+                        flatten_reqs.push(self.sstable_store.sstable_syncable(sst, &stats));
+                    }
+
+                    let flatten_resp = futures::future::try_join_all(flatten_reqs).await?;
+                    let mut sstables: Vec<TableHolder> = Vec::with_capacity(flatten_resp.len());
+                    for (sst, _) in flatten_resp {
+                        sstables.push(sst);
+                    }
+                    sstables.sort_by(|a, b| {
+                        KeyComparator::compare_encoded_full_key(
+                            &a.value().meta.smallest_key,
+                            &b.value().meta.smallest_key,
+                        )
+                    });
+                    for idx in 1..sstables.len() {
+                        if KeyComparator::compare_encoded_full_key(
+                            &sstables[idx].value().meta.smallest_key,
+                            &sstables[idx - 1].value().meta.largest_key,
+                        ) != std::cmp::Ordering::Greater
+                        {
```

Comment:
It means that it is an overlapping level. In our code the output level would never be overlapping-level, but I add a check here because if the output level is not sorted-by-key, the algorithm in this function would be wrong.

Reply:
> It means that it is an overlapping level. In our code the output level would never be overlapping-level, but I add a check here because if the output level is not sorted-by-key, the algorithm in this function would be wrong.

Therefore I think we should panic here and should not return an `Ok(())`